core/lib/basic_types/src/tee_types.rs

```diff
@@ -1,9 +1,58 @@
+use std::{fmt, str::FromStr};
+
 use serde::{Deserialize, Serialize};
-use strum::{Display, EnumString};
 
-#[derive(Debug, Clone, Copy, PartialEq, EnumString, Display, Serialize, Deserialize)]
+#[derive(Debug, Clone, Copy, PartialEq, Serialize, Deserialize)]
+#[serde(rename_all = "lowercase")]
 #[non_exhaustive]
 pub enum TeeType {
-    #[strum(serialize = "sgx")]
     Sgx,
 }
+
+impl fmt::Display for TeeType {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            TeeType::Sgx => write!(f, "sgx"),
+        }
+    }
+}
+
+impl FromStr for TeeType {
+    type Err = String;
```

Comment:
`String` is not a good error type. `anyhow::Error` would work better here.

Reply:
I completely removed the implementation of the std::str::FromStr trait because we don't use it anyway.